### PR TITLE
[BUGFIX] Fix comment parsing to support multiple comments (backported)

### LINKED ISSUE #173: bug with parsing multiple comments
When parsing a rule or csslist, trailing comments (those comments after the rule/list) are consumed as part of a final consumeWhitespace call, meaning we don't see these reported for the next rule in the set (or the next set).

Consider the following cases:

`div {/*Find Me!*/left:10px; /*Find Me Too!*/text-align:left;}`
Both of these comments should be returned (one for each rule), but the latter is not.

`/*Find Me!*/div {left:10px; text-align:left;} /*Find Me Too!*/a {left:10px;}`
Both of these comments should be returned (one for each csslist), but the latter is not.

I'd like this fixed so I can use this to parse RTL directives for css rules/lists reliably again. Moodle recently upgraded to 8.3.0 and this broke our RTL.

Please see the linked fix proposal.

### PR DESCRIPTION
**Pull request by ziegenberg**
Because of an eager consumption of whitespace, the rule parsing would swallow a trailing comment, meaning the comment for the next rule would be affected. This patch addresses this by only consuming real whitespace without comments after a rule.

Fixes #173

### CODE REVIEW

**Review by oliverklee — APPROVED** [review]
Thanks!

**Comment by oliverklee** [comment]
I'm a bit confused … do I need to forward-port this, or is this not necessary due to other upcoming changes in `main`?

(In general, if we want to create a PR that we are also planning to backport, we first create the PR for `main` and then backport it to the maintenance branch after it has been merged.)

**Comment by oliverklee** [comment]
I've now forward-ported this in #672. For future patches, please let's create the change for `main` first, and I (or another maintainer) will then take care of the backport. (This also works better with the branch protection in place, where we need another reviewer if we want to merge to `main` and would need to override this protection if we want to do a forward-port without an additional review.)

**Comment by ziegenberg** [comment]
I wasn't done with the patches. As the solution for v8.x looks very different from the proposed solution for v9.0, I was about to create different patches. Either back-porting or forward-porting would have meant a completely different approach.

**Comment by oliverklee** [comment]
All right. I’m looking forward to your PRs! 🙏

**Comment by oliverklee** [comment]
And if you think talking things through would be helpful, please feel free to chat me up.

**Pull request merged**
